deployment/README.md

```diff
@@ -1,8 +1,89 @@
-# Build scripts
+# Jellyfin Packaging
 
-All `build.sh` and `package.sh` scripts are for *nix platforms (or WSL on Windows 10).
+This directory contains the packaging configuration of Jellyfin for multiple platforms. The specification is below; all package platforms must follow the specification to be compatable with the central `build` script.
 
-After running both, check the `*/pkg-dist/` folders for the archives and packages.
+## Package List
+
+### Operating System Packages
+
+* `debian-package-x64`: Package for Debian and Ubuntu amd64 systems.
+* `fedora-package-x64`: Package for Fedora, CentOS, and Red Hat Enterprise Linux amd64 systems.
+
+### Portable Builds (archives)
+
+* `debian-x64`: Portable binary archive for Debian amd64 systems.
+* `ubuntu-x64`: Portable binary archive for Ubuntu amd64 systems.
+* `linux-x64`: Portable binary archive for generic Linux amd64 systems.
+* `osx-x64`: Portable binary archive for MacOS amd64 systems.
+* `win-x64`: Portable binary archive for Windows amd64 systems.
+* `win-x86`: Portable binary archive for Windows i386 systems.
+* `win-generic`: Portable binary archive for generic Windows systems.
+
+### Framework Builds
+
+* `framework`: Compiled `.dll` for use with .NET Core runtime on any system.
+
+## Package Specification
+
+### Action Scripts
+
+* Actions are defined in BASH scripts with the name `<action>.sh` within the framework directory.
+
+* The list of valid actions are:
+
+    1. `build`: Builds a set of binaries.
+    2. `package`: Assembles the compiled binaries into a package.
+    3. `sign`: Performs signing actions on a package.
+    4. `publish`: Performs a publishing action for a package.
+    5. `clean`: Cleans up any artifacts from the previous actions.
+
+* All package actions are optional, however at least one should generate output files, and any that do should contain a `clean` action.
+
+* Actions are executed in the order specified above, and later actions may depend on former actions.
+
+* Actions except for `clean` should `set -o errexit` to terminate on failed actions.
+
+* The `clean` action should always `exit 0` even if no work is done or it fails.
+
+### Output Files
+
+* Upon completion of the defined actions, at least one output file must be created in the `<platform>/pkg-dist` directory.
+
+* Output files will be moved to the directory `jellyfin-build/<platform>` one directory above the repository root upon completion.
+
+### Common Functions
+
+* A number of common functions are defined in `deployment/common.build.sh` for use by platform scripts.
+
+* Each action script should import the common functions to define a number of standard variables.
+
+#### `get_version`
+
+Reads the version information from `SharedVersion.cs`.
+
+**Arguments:** `ROOT`
```

Comment:
why here's no doc on arguments' meaning?

Reply:
Cause I was lazy, and I did a thing of "name should speak for itself."

Reply:
They're reallt consistent though (thanks @EraYaN) so we should just need to explain each one once.

Reply:
Added explanations.